apps/angular/signal-input/src/app/user.component.ts

```diff
@@ -2,40 +2,32 @@ import { TitleCasePipe } from '@angular/common';
 import {
   ChangeDetectionStrategy,
   Component,
-  Input,
-  OnChanges,
+  computed,
+  input,
 } from '@angular/core';
+import { Category, CategoryType } from './models/category.model';
 
-type Category = 'Youth' | 'Junior' | 'Open' | 'Senior';
-const ageToCategory = (age: number): Category => {
-  if (age < 10) return 'Youth';
-  else if (age < 18) return 'Junior';
-  else if (age < 35) return 'Open';
-  return 'Senior';
-};
+const CATEGORY_DEFAULT: CategoryType = 'JUNIOR';
 
 @Component({
   selector: 'app-user',
   standalone: true,
   imports: [TitleCasePipe],
   template: `
-    {{ fullName | titlecase }} plays tennis in the {{ category }} category!!
+    {{ fullName() | titlecase }} plays tennis in the {{ category() }} category!!
   `,
   host: {
     class: 'text-xl text-green-800',
   },
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class UserComponent implements OnChanges {
-  @Input({ required: true }) name!: string;
-  @Input() lastName?: string;
-  @Input() age?: string;
+export class UserComponent {
+  name = input.required<string>();
+  lastName = input<string>();
+  age = input(undefined, { transform: (v: string) => Number(v) });
 
-  fullName = '';
-  category: Category = 'Junior';
-
-  ngOnChanges(): void {
-    this.fullName = `${this.name} ${this.lastName ?? ''}`;
-    this.category = ageToCategory(Number(this.age) ?? 0);
-  }
+  fullName = computed(() => `${this.name()} ${this.lastName()}`);
+  category = computed(() =>
+    this.age() ? Category.fromAge(this.age()!) : CATEGORY_DEFAULT,
```

Comment:
@tomalaforge I changed the function name here, only to show why i like to use that static method in a class approach. It is more boilerplate but it reads like a Sentence. Get the `Category.fromAge(this.age()!)`.

Reply:
If you prefer it and you feel more comfortable with it. that's good. the goal is to have a scalable and maintainable application so that you can easily refactor it and understand it in a couple of months.
I know that i prefer functional programming, but that's my personal preference. In the end, there is no gain in performance.